app/Http/Controllers/Category/IndexController.php

```diff
@@ -54,6 +54,10 @@ function ($request, $next) {
                 return $next($request);
             }
         );
+
+        // todo This is just a placeholder, for now (although it's likely this feature will, in fact,
+        //  be part of the basic package)
+        $this->middleware('feature:basic');
```

Comment:
This is the current proposal for implementing feature-limiting, where we have a fairly simple middleware, that handles checking the current user against a certain feature level. As feature levels are on a vertical progression, any user below the specified level is blocked, while any at, or above, is allowed to access.
One issue here, is that we don't know what level is required for access, before attempting to access the feature, meaning it's also difficult to filter out, or mark, the navigation pane, for features that the user does not have access to. We're also stuck with a fairly generalised error message, which can potentially be customized (depending on the support within middlewares, that's not investigated currently), but otherwise is just a simple "You don't have access to this feature".